File: leetcode/binary_search/1060_Missing_Element_in_Sorted_Array/codes/bs.py
from typing import List

class Solution:
    def missingElement(self, nums: List[int], k: int) -> int:
        def missing(i):
            return nums[i] - nums[0] - i

        n = len(nums)
        if missing(n - 1) < k:
            return nums[-1] + k - missing(n - 1)

        left, right = 0, n - 1
        while left < right:
            mid = (left + right) // 2
            if missing(mid) < k:
                left = mid + 1
            else:
                right = mid
        return nums[left - 1] + k - missing(left - 1)

File: leetcode/binary_search/1060_Missing_Element_in_Sorted_Array/codes/bs.c
int missingElement(int* nums, int numsSize, int k) {
    int missing;

    int getMissing(int i) {
        return nums[i] - nums[0] - i;
    }

    if (nums[numsSize - 1] - nums[0] - (numsSize - 1) < k) {
        return nums[numsSize - 1] + k - (nums[numsSize - 1] - nums[0] - (numsSize - 1));
    }

    int left = 0, right = numsSize - 1;
    while (left < right) {
        int mid = (left + right) / 2;
        missing = nums[mid] - nums[0] - mid;
        if (missing < k) {
            left = mid + 1;
        } else {
            right = mid;
        }
    }

    missing = nums[left - 1] - nums[0] - (left - 1);
    return nums[left - 1] + k - missing;
}
